### DISCUSSION TITLE
How do I check if every song has a highscore

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [CodenameCrew/CodenameEngine#464] [Q&A]
Tryna make a cool ImpostorV4 type thing where, after setting a highscore for every song, it opens a fake menu
anyway I need to figure this out and I kinda just wanna use an onSongEnd function
if that's not possible or lucrative, I'm open to other options

**Comment by r6915ee** [comment]
> Tryna make a cool ImpostorV4 type thing where, after setting a highscore for every song, it opens a fake menu anyway I need to figure this out and I kinda just wanna use an onSongEnd function if that's not possible or lucrative, I'm open to other options

How does Impostor4 do such? I don't think I remember something like that being used there.

**Reply by THEkatinamicrowave** [reply]
I believe that after you beat all the story mode songs, it like plays the Endgame cutscene and sends you to a fake freeplay menu where you can only play the boss fight
idk man play Impostor 4 yourself

**Reply by r6915ee** [reply]
> I believe that after you beat all the story mode songs, it like plays the Endgame cutscene and sends you to a fake freeplay menu where you can only play the boss fight idk man play Impostor 4 yourself

Ah, THAT scenario is what you meant. I've played Impostor4, I just didn't exactly understand what you meant in the original post.

One idea for me that comes to mind is to change the related song list files in your `FreeplayState` before they actually get displayed for the first time. If you have no custom states, then I would probably use a script that activates upon `FreeplayState` being loaded. Notable variables would be `songs`, `songList` and `iconArray`, though I don't exactly know how they're used.
